### PR TITLE
Doc fixes for upstream numpy/astropy

### PR DESCRIPTION
**Pull request by nabobalis**
These should fix the issues with the cronjob under dev Astropy/NumPy.

I also split up the cron job to have them separately as well as a check. There is little issue with a few extra pointless cronjob tests. 

I also fix an issue (hopefully, @Cadair you check) with VSO showing up twice. I haven't checked since my local doc build is broken for an unknown reason tho suspected reason.

### CODE REVIEW

File: docs/code_ref/net.rst [hunk 1]
```diff
@@ -34,9 +34,6 @@ VSO
 .. automodapi:: sunpy.net.vso
    :headings: ^#
 
-.. automodapi:: sunpy.net.vso.vso
```
Comment (dstansby) [inline docs/code_ref/net.rst]: This is the missing bit of the docs I added back in recently! Is there a reason it's going again?

Comment (dstansby) [reply docs/code_ref/net.rst]: (xref #2353)

Comment (nabobalis) [reply docs/code_ref/net.rst]: Yeah, sorry about that. We had some of the vso API already imported in the vso/__init__ file so we had a duplicate in the docs search. I added the vso ones into the __init__ so it should be there now. I might have overlooked some of the other ones. I will check.

Comment (Cadair) [reply docs/code_ref/net.rst]: @dstansby can you remember the objects that were missing? as far as I could tell it was only `QueryResponse` which this PR moves to the `sunpy.net.vso` namespace.

Comment (dstansby) [reply docs/code_ref/net.rst]: It was `QueryResponse` and `VSOClient` - which appear in the devdocs here: http://docs.sunpy.org/en/latest/code_ref/net.html#module-sunpy.net.vso.vso (but I guess `VSOClient` was duplicated)

(I'm also very confused by documentation for code that lives in `net.vso.vso` ending up under in `net.vso` package documentation)

Comment (nabobalis) [reply docs/code_ref/net.rst]: You mean that to use the functions under `net.vso.vso`, you can import them under the `net.vso` namespace?

Comment (dstansby) [reply docs/code_ref/net.rst]: Ah, so if I import `net.vso` I get the stuff in `net.vso.vso` too. Got it!

Comment (Cadair) [reply docs/code_ref/net.rst]: You get everything defined in `net/VSO/__init__.py` which with this PR incluses `QueryResponse` and `VSOClient` imported from `vso.vso`. Which now means both of those are documented as members of `sunpy.net.vso`

File: sunpy/net/vso/vso.py [hunk 2]
```diff
@@ -966,4 +964,4 @@ def get(query_response, path=None, methods=('URL-FILE',), downloader=None):
     return g_client.get(query_response, path, methods, downloader)
 
 
-get.__doc__ = VSOClient.get.__doc__
+get.__doc__ = VSOClient.search.__doc__
```
Comment (dpshelio) [inline sunpy/net/vso/vso.py]: Why changing the `get` doc to the `search` one?

File: sunpy/net/vso/__init__.py [hunk 3]
```diff
@@ -6,4 +6,4 @@
 # Delete this in 0.9
 from sunpy.net.vso.vso import search, get
 
-__all__ = ['VSOClient', 'InteractiveVSOClient']
+__all__ = ['VSOClient', 'InteractiveVSOClient', 'QueryResponse', 'search', 'get']
```
Comment (Cadair) [inline sunpy/net/vso/__init__.py]: I don't think `search` and `get` should be added here, they are deprecated anyway.

File: sunpy/net/vso/vso.py (hunk 2 above)
Comment (Cadair) [reply sunpy/net/vso/vso.py]: because `VSOClient.get` now has no actual docs, because it's now deprecated.

**Pull request merged**